src/components/Contacts/MassActions/AddTags/MassActionsAddTagsModal.tsx

```diff
@@ -70,6 +70,38 @@ export const MassActionsAddTagsModal: React.FC<
     },
   });
 
+  const handleValidation = async (fields: Partial<ContactUpdateInput>) => {
+    const contactsExistingTags =
+      contactsForTags?.contacts.nodes.map((contact) => ({
+        tagList: [...new Set([...contact.tagList])],
+      })) ?? [];
+
+    const existingTags = [];
+    contactsExistingTags.forEach((contact) => {
+      for (let i = 0; i < contact.tagList.length; i++) {
+        existingTags.push(contact.tagList[i]);
+      }
+    });
+
+    for (let i = 0; i < fields.tagList?.length; i++) {
+      existingTags.push(fields.tagList[i]);
+      const duplicates = existingTags.filter(
+        (item, index) => existingTags.indexOf(item) !== index,
+      );
+      if (duplicates.length > 0) {
+        enqueueSnackbar(
+          t('One or more selected contacts already has this tag'),
```

Comment:
@dr-bizz I think this should only prevent the user from adding tags if **every** selected contact already has the tag. If one or 2 of them already have the tag, I would want it to add the tag to the rest of the contacts and just skip over the ones that already have it. What do you think?

For example, if I've added a `likesBaseball` tag to a couple of contacts already and the next week I want to mass select and add it to 10 more people, I would have to go back to each of the 10 contacts and make sure I haven't already added the tag.

Reply:
Agreed.